crawler_mcp/crawlers/web.py

```
@@ -1088,6 +905,240 @@ async def _fetch_text(self, url: str, timeout: int = 10) -> str:
+        # Limit sitemap URLs to max_pages
+        urls_to_crawl = sitemap_urls[:max_pages]
+
+        # Check if MemoryAdaptiveDispatcher is available
+        if MemoryAdaptiveDispatcher is None:
+            logger.warning(
+                "MemoryAdaptiveDispatcher unavailable, falling back to sequential crawling"
+            )
+            # Fallback to sequential crawling
+            for url in urls_to_crawl:
+                try:
+                    result = await crawler.arun(url=url, config=run_config)
```

Ruff (0.12.2)

Undefined name `crawler`

(F821)